app/src/main/java/org/summerofbitcoin/wallet/utilities/SharedPrefManager.kt

```diff
@@ -0,0 +1,46 @@
+package org.summerofbitcoin.wallet.utilities
+
+import android.content.SharedPreferences
+
+class SharedPrefManager(private val sharedPreferences: SharedPreferences) {
+
+    var path: String
+        get() = sharedPreferences.getString(PREFS_PATH, "") ?: ""
+        set(value) {
+            sharedPreferences.edit()?.putString(PREFS_PATH, value)?.apply()
+        }
+
+    var descriptor: String
+        get() = sharedPreferences.getString(PREFS_DESCRIPTOR, "") ?: ""
+        set(value) {
+            sharedPreferences.edit()?.putString(PREFS_DESCRIPTOR, value)?.apply()
+        }
+
+    var changeDescriptor: String
+        get() = sharedPreferences.getString(PREFS_CHANGE_DESCRIPTOR, "") ?: ""
+        set(value) {
+            sharedPreferences.edit()?.putString(PREFS_CHANGE_DESCRIPTOR, value)?.apply()
+        }
+
+    var mnemonic: String
+        get() = sharedPreferences.getString(PREFS_MNEMONIC, "No seed phrase saved")
+            ?: "Seed phrase not there"
+        set(value) {
+            sharedPreferences.edit()?.putString(PREFS_MNEMONIC, value)?.apply()
+        }
+
+    var walletInitialised: Boolean
+        get() = sharedPreferences.getBoolean(PREFS_WALLET_INITIALISED, true)
```

Comment:
This should have a default value of `false`. The app otherwise crashes upon startup.

Reply:
Oops, my build was running on the old sharedPrefs values, hence it did not crash for me :sweat_smile: . I reset my storage and cache and was able to reproduce this issue. I have sent a fix for the same :smiley: